x-pack/docs/en/security/authentication/jwt-realm.asciidoc

```diff
@@ -12,35 +12,61 @@ between the _client_ that is connecting to {es}, and the _user_ on whose behalf
 the request should run. The JWT authenticates the user, and a separate credential
 authenticates the client.
 
-A common scenario for JWTs is when an existing front-end application uses
-OpenID Connect (OIDC) to authenticate and identify a user, and then accesses {es}
-on behalf of the authenticated user.
-
-TIP: If the front-end application does not exist, you can use the
-<<token-authentication-services>> instead.
+The JWT realm is designed to cater for the following two scenarios:
+
+1. An application authenticates and identifies a user with an authentication flow, e.g.
+OpenID Connect (OIDC), and then accesses {es} on behalf of the authenticated user.
+A common scenario for JWTs is when an existing application uses
+2. An application itself goes through an authentication flow, e.g. OAuth2 Client Credentials
+Flow, and then accesses {es} for itself.
+
+In both scenarios, the application should present a JWT to {es} that represents either the user (scenario 1)
+or itself (scenario 2). {es} categorizes the JWT into two types, ID Token and Access token,
+and validates them accordingly. The ID Token type basically follows the OIDC specification.
+The access token type is similar to the ID Token type, but with more relaxed rules which makes it
+suitable for loosely-defined JWTs, including self-signed ones.
+Tokens of both types must contain the following 5 pieces of information. While ID Tokens follow
+strict rules on which claim should provide each piece of the information,
+access tokens allow certain configurable options.
```

Comment:
I think we do a disservice by trying to describe both token types together. We debated having 2 different realm types for ID token and Access token because they are logically quite different. I think having the docs reflect that difference might be more helpful - it means readers are force to think through their use case and pick the right option rather than think about it as simply being different configuration options on the same basic authc method.

Reply:
I intentionally tried to compare them side by side. I feel their similiarities are much more than the differences. But I agree we should encourage readers to think their use case more and choose carefully. I rephrased this section with these in mind.